modules/lang-painless/src/main/java/org/elasticsearch/painless/api/Augmentation.java

```diff
@@ -712,4 +715,13 @@ public static Matcher matcher(Pattern receiver, int limitFactor, CharSequence in
         }
         return receiver.matcher(new LimitedCharSequence(input, receiver, limitFactor));
     }
+
+    /**
+     * Convert a {@link TemporalAccessor} into millis since epoch like {@link Instant#toEpochMilli()}.
+     */
+    public static long toEpochMilli(TemporalAccessor v) {
+        long millis = Math.multiplyExact(v.getLong(ChronoField.INSTANT_SECONDS), 1_000);
+        millis = Math.addExact(millis, v.get(ChronoField.NANO_OF_SECOND) / 1_000_000);
+        return millis;
```

Comment:
No suggestions here. Just surprised this is the best way to do it :).

Reply:
Its not exactly the clearest thing, no. If you *know* that you have a `ZonedDateTime` you can do `zdt.toInstant().toEpochMilli()` which does about the same thing but you have to worry about if the JVM can skip allocating the `Instant`.... And we're not sure we have a `ZonedDateTime` here either.

Reply:
Makes sense. Thanks for the explanation!

Reply:
Why the `Math.` calls instead of actual math?